assets/js/components/DashboardNavigation.js

```diff
@@ -99,15 +100,36 @@ export default function DashboardNavigation() {
 
 	const breakpoint = useBreakpoint();
 
-	const [ selectedIds, setSelectedIds ] = useState( [
-		global.location.hash.substr( 1 ),
-	] );
+	const [ selectedID, setSelectedID ] = useState(
+		global.location.hash.substr( 1 )
+	);
+
+	/**
+	 * A race condition occurs between the onScroll
+	 * throttle function and the handleSeclect callback.
+	 * This ref value will be set as false in the handleSelect callback.
+	 * Hence, setting the selectedID and updating the URL hash
+	 * won't be exceuted always except the screen is scrolled
+	 * to the appropriate section. This prevents the race condition issue.
+	 */
+	const shouldScroll = useRef( true );
 
 	const handleSelect = useCallback(
-		( selections ) => {
-			const [ hash ] = selections;
+		( chipID ) => {
+			const hash = chipID;
 			if ( hash ) {
-				global.history.replaceState( {}, '', `#${ hash }` );
+				shouldScroll.current = false;
+
+				/**
+				 * This sets the #traffic anchor when clicking the `Traffic`
+				 * chip for the first time when there is no hash in the URL.
+				 * Setting the anchors afterwards will occur in the onScroll function.
+				 */
+				const context = getContextScrollTop( `#${ hash }`, breakpoint );
+				if ( context < 5 ) {
+					global.history.replaceState( {}, '', `#${ hash }` );
+					setSelectedID( hash );
```

Comment:
I think we don't need to set the selected ID in the `handleSelect` callback anymore because it will be handled in the onScroll event. We can use this `handleSelect` function only to scroll the page to a certain area. Hence the `shouldScroll` isn't needed.

Also, we don't need this function to be a callback anymore and can remove the `useCallback` wrapper.